Security Considerations for Unsigned Credential ID

selfissued opened the pull request:
Fixes #394


<!--
    This comment and the below content is programatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/selfissued/webauthn/pull/766.html" title="Last updated on Jan 26, 2018, 4:21 PM GMT (86f8f7c)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/766/1e77b42...selfissued:86f8f7c.html" title="Last updated on Jan 26, 2018, 4:21 PM GMT (86f8f7c)">Diff</a>

emlun commented on index.bs:
```diff
@@ -4427,7 +4427,13 @@ RogueList when performing ECDAA-Verify (see section 3.6 in [[!FIDOEcdaaAlgorithm
+## credentialId Unsigned ## {#credentialIdSecurity}
 
+The credential ID is not signed.
+This is not a problem because all that would happen if an authticator returns
```
Typo: authticator

emlun commented on index.bs:
```diff
@@ -4427,7 +4427,13 @@ RogueList when performing ECDAA-Verify (see section 3.6 in [[!FIDOEcdaaAlgorithm
 
+The credential ID is not signed.
+This is not a problem because all that would happen if an authticator returns
+the wrong crednetial ID is that the RP will not look up the correct
```
Typo: crednetial
Suggest: RP -> [=[RP] =]

emlun replied:
I think a more interesting case to point out is if a MitM were to manipulate the credential ID (the conclusion is the same), so I would probably write this as something like "[...] if an authenticator returns the wrong credential ID, or if an attacker intercepts and manipulates the credential ID, is that the RP [...]"

emlun commented:
Also, perhaps link some of the terms throughout? I can make a PR of that into this PR if you can't run bikeshed.

selfissued commented:
@emlun if you can create a PR into my PR addressing the issues you raise that would great.  Then I could apply it in my morning and merge the combination into master.
